keymapper/injection/consumers/keycode_mapper.py

```diff
@@ -430,22 +433,41 @@ def handle_keycode(self, event, action, forward=True):
                 unreleased_entry = unreleased[type_and_code]
                 target_type, target_code = unreleased_entry.target_type_code
                 del unreleased[type_and_code]
-
+                
+                target_uinput = None
+                if isinstance(target_code, tuple):
+                    target_uinput = target_code[1]
+                    target_code = target_code[0]
```

Comment:
```suggestion
                if isinstance(target_code, tuple):
                    target_code, target_uinput = target_code
```

When would target_code not be a tuple with the changes to `_map_keys_to_codes`?

Reply:
Forwarded keycodes don't have a `target_uinput`, only a `target_code`

Reply:
Can we use `None` in `target_uinput` in that case? To have a consinstent data structure. Based on experience, handling different formats leads to bugs.

Reply:
Maybe it helps to establish a new class for this purpose

```py
class Target:
    def __init__(uinput, event_type, code, value):
        ...
```

You could use this in `_map_keys_to_codes`, and in `target_type_code` of `Unreleased`. That way there is a clear definition.

(since `type` is a builtin keyword, it need to be suffixed)

Reply:
Okay, had a look how I might implement the Idea, but it does not seem worth the effort. I chose `tuple` on purpose because it is immutable if we replace it with a Target object we have to be careful not to mess up the mapping by accident. 
But I will keep the idea in mind if I end up refactoring `keycode_mapper`.

The changes I have done so far seem to work well. You can test them by changing a preset to something like this: 
```
{
    "mapping": {
        "1,3,1": ["BTN_A","gamepad"],
        "1,2,1": ["d","keyboard"],
        "1,4,1": ["c","keyboard"],
        "1,5,1+1,6,1": ["a","keyboard"],
        "1,7,1": ["b+BTN_2","keyboard"]
    }
}
```
And load it with `key-mapper-control` the GUI is probably broken.

Reply:
For immutability, I think making `def __setattr__()` throw an error would work.

I'll test it this weekend, thanks